collection/rancher/v2.x/logs-collector/rancher2_logs_collector.sh

```diff
@@ -153,6 +160,21 @@ sherlock-data-dir() {
 
 }
 
+verify-kubeconfig() {
```

Comment:
Sorry, what's the need for this? Why introduce another flag if the `KUBECONFIG` variable is already pointed to the desired location?

Reply:
`verify-kubeconfig` function does 2 things:

-  verifies the file specified with `-k` if it exists and it's non-zero
-  if not sets KUBECONFIG to a default user location ~/.kube/config

i think this is sufficient to cover the most common cases, since sudo is required.

the variable SUDO_USER sets it to the user sudo is ran as so that ~/.kube/config is searched in the expected place when the script is executed. i have tested this from my kubeadm cluster and it works.

Using a function (`verify-kubeconfig`) has several advantages IMO, it can be reused elsewhere in the code when needed and also it can expanded later on adding more complex checks.
i think it could also benefit other collections like k3s, rke2 and so on being able to specify a custom file...